### PR TITLE
Add navigation commands into the editor

### PR DESCRIPTION
**Pull request by ryanhoangt**
## Description

This PR is to add navigation commands into the editor:
- `jump_to_definition`
- `find_references`

The 2 commands will be enabled when detecting a git repo in cwd. Otherwise, it will print:

```
No git repository found. Navigation commands are disabled. Please use bash commands instead.
```

Also, if the navigator commands are enabled, when the agent `view`s a file, the content is also appended with <TIP> to encourage exploration. If no git repo found the TIP will not be appended.

```
<TIP>Use the `jump_to_definition` and `find_references` commands to understand more about a particular class, function/method and how it is used in the codebase.</TIP>
```

## How Has This Been Tested?

https://github.com/user-attachments/assets/9f033b6b-9c1d-4262-9c78-74f35d34eb80

### CODE REVIEW

**Comment by ryanhoangt** [comment]
Hey @xingyaoww, I think this PR is mostly ready. Some remaining issues I'm not sure how to handle properly yet:
- The `tree-sitter` parser uses a `.oh_aci.cache.tags` directory for caching purpose, which means it'll create this directory in the workspace. This might be a bit annoying, and I'm thinking about removing this cache dir, as the command itself is not used very frequently by the agent. But with repomap we should find some ways to deal with this, as the repomap is constructed for every step of execution.
- In the `GitRepoUtils` class, I'm doing some traversal to detect the git repo in the workspace. But if under `workspace` there're multiple git repos, the current implementation may pick up the wrong one.
Do you have some suggestions on these?

**Comment by xingyaoww** [comment]
> The tree-sitter parser uses a .oh_aci.cache.tags directory for caching purpose, which means it'll create this directory in the workspace. This might be a bit annoying, and I'm thinking about removing this cache dir, as the command itself is not used very frequently by the agent. But with repomap we should find some ways to deal with this, as the repomap is constructed for every step of execution.

Is there a way for us to create this directory elsewhere? Say something under `/tmp`?

> In the GitRepoUtils class, I'm doing some traversal to detect the git repo in the workspace. But if under workspace there're multiple git repos, the current implementation may pick up the wrong one. Do you have some suggestions on these?

Hmm interesting, can you point me to the exact code for this? I'm thinking we could potentialy track back from an model given path?
Say the model want to view `/workspace/a/b/c.py` - we first check if `b` is a git repo, if not, we go to `a`?

**Comment by ryanhoangt** [comment]
@xingyaoww I figured out a way to handle the workspace thing, thanks to the pwd already set to bash pwd for ipython. This should be finally ready. Please take another look, then I'll cut a release and finish up the PR in OH repo.

File: openhands_aci/tree_sitter/parser.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        if not code:
+            return []
+
+        parsed_tree = ts_parser.parse(bytes(code, 'utf-8'))
+
+        # Run the tags queries
+        query = ts_language.query(tags_query)
+        captures = query.captures(parsed_tree.root_node)
+
+        parsed_tags = []
```
Comment (xingyaoww) [inline openhands_aci/tree_sitter/parser.py:73]: Why do you call it `tags`? Seems a bit confusing? Is it just `node`?

Comment (ryanhoangt) [reply openhands_aci/tree_sitter/parser.py:73]: It's a term in `tree-sitter` when using query files, the output can be called tag, which is just a named entity, or syntax node: https://tree-sitter.github.io/tree-sitter/code-navigation-systems

File: openhands_aci/editor/navigator.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+        # Caching
+        self.file_context_cache: dict = {}  # (rel_file) -> {'context': TreeContext_obj, 'mtime': mtime})
+        self.rendered_tree_cache: dict = {}  # (rel_file, lois, mtime) -> rendered_tree
+
+    @property
+    def git_utils(self):
+        if self._git_repo_found is None:
+            pwd = os.getcwd()
+            if pwd == '/testbed':
```
[the remark this answers is not in the document]
Comment (ryanhoangt) [reply openhands_aci/editor/navigator.py]: We're gonna remove the symlink in swe-bench eval so I'll remove this code

**Comment by xingyaoww** [comment]
lmk when this is ready for another look!

**Comment by ryanhoangt** [comment]
Close this in favor of https://github.com/All-Hands-AI/openhands-aci/pull/35.

**Pull request closed without merging**